### DISCUSSION TITLE
Locations of diagnostics getting out of sync

### DISCUSSION THREAD

**Discussion by deifactor** [joaotavora/eglot#1047] [Q&A]
If I open a Rust file with the following contents:

```rust
fn main() {
    let x = 1;
    println!("{}", x.baz);
}
```

and then add a bunch of newlines above the `println!`, the error diagnostic shifts from `x.baz` to earlier in the line:

![image](https://user-images.githubusercontent.com/30192992/191160877-ae642c7d-7501-41e4-aa04-8c80fae8bac2.png)

Notably, if I set `(setq flymake-no-changes-timeout 1000)`, then everything is fine until I `flymake-start`. It looks like something in the LSP <-> eglot <-> flymake chain is messing up the positions.

Saving the file causes it to snap back to the right place.

**Comment by tshu-w** [comment]
Some related [discussion](https://github.com/joaotavora/eglot/discussions/1083) and [issue](https://github.com/joaotavora/eglot/issues/1296#issuecomment-1727978307).
